### DISCUSSION TITLE
[Bug]: Fixed GPS coordinates not retained, RAK4631

### DISCUSSION THREAD

**Discussion by dhchenault** [meshtastic/firmware#5201] [Troubleshooting]
### Category

BLE, Serial, Other

### Hardware

Rak4631

### Firmware Version

2.5.4.8d288d5

### Description

My node won't keep the fixed GPS coordinates. Using RAK4631 without a GPS in Router mode, v.2.5.4., GPS=NOT_PRESENT, Smart Position= Off, GPS update set super large ("999999"), Fixed Location=ON. It takes GPS from my android phone while connected via BT, but when remotely configuring it from another node, the GPS coordinates are gone and doesn't appear in the map. Docs seem to indicate for GPS updates, it should reuse the last saved coordinates.

**Comment by garthvh** [comment]
Fixed position requires a specific order, you set the position, turn on fixed and the position is retained until you turn off fixed position or clear your nodedb.  Did you set the position before setting the bool, or have you been clearing your node db?

**Comment by dhchenault** [comment]
No, I set the coordinates after setting the switch for fixed position. Have not cleared the DB in some time. I will try that sequence.

**Reply by bmazor-uofa** [reply]
ive got same issue. so frustrating

**Comment by dhchenault** [comment]
I turned off fixed and rebooted. Then went back but the coordinates fields don't appear until toggling the Fixed Position on. However, when it toggles on, the correct coords are there, pulled from my phone. Send and reboot. Looks good. But when I disconnect and connect to a Client node then remote config the first, coords are "0" again. 
![Screenshot_20241031_135213_Meshtastic](https://github.com/user-attachments/assets/aa3758dd-94cb-4822-a607-411b71e90771)
![Screenshot_20241031_135208_Meshtastic](https://github.com/user-attachments/assets/3b87061d-1de7-4f09-8778-bcec6fa0f1d6)

**Reply by dhchenault** [reply]
So I've tried toggling the Fixed Position on from both Android and iOS apps. Neither result in GPS coordinates being stored. In the iOS app, there is no position log (grayed out).

**Reply by garthvh** [reply]
Must not have a gps lock

**Reply by ikearat** [reply]
He's using "Not-present" and "Fixed" the GPS is not involved. I have a similar issue with a Heltec V3. the Position feature of Meshtastic is creepy.

**Reply by dhchenault** [reply]
According to seperate GPS app running at the same time on the mobile device, there is a lock with 10+ sats. With the Android, the coordinates show up in the fields. After reboot, the appear correctly with the node in the list of nodes. But, when I connect to a second node and view nodes, there is no position reported and node is not on the map. I can request position and still nothing. If I'm reading it right, debug packets report empty position field. When I remote configure the original node from the second node, the coordinates field in the Position settings show "0.0".

**Reply by ikearat** [reply]
Quick question: In the Radio configuration select Channels and touch the zero channel... is the Position enabled item true or false? Is the slider to the right and "lit"?
Also, reduce the position broadcast interval to 20 seconds or something or you wait the 43200 seconds for an update to the position data.

**Comment by drmzperx** [comment]
The same issue with Rak4631 ETH+POE in client mode: the fixed position coordinates are not broadcasted to MSH and MQTT as well. All channels are precise position enabled. Unfortunately, the node is in an inaccessible (high) location, so I cannot check the serial output in the near future.
Firmware: 2.5.7 -> 2.5.11 all.

**Reply by garthvh** [reply]
You can't subscribe to the root msh topic

**Reply by drmzperx** [reply]
You mean mqtt msh/ root topic? Yes I know, I use private mqtt server with custom topics.

**Comment by dtormanen** [comment]
My similar rak4631 problem was as follows. After upgrading firmware to 2.5.14, I noticed my position on the map was off.  Once connected via BT, it tried to set the fixed position back to where I wanted it. After reboot, the position would show correct for a few seconds then bounce over to the incorrect position. I shut down BT connection and connected serially via https://client.meshtastic.org and Chrome browser. I tried setting fixed position via serial interface but entering fixed position lat/lon in browser serial connection isn't available. I noticed in Channels, Primary, Precise location was set to off.  Once setting precise location to on, or enabled, the rak4631 position would stay in place on the map.  Apparently, with Precise Location turned off, my entered set fixed position was obfuscated.

**Reply by bmazor-uofa** [reply]
THIS FIXED IT!!!! THANK YOU!!

Radio Configuration >> Channels >> click whatever you have, i have "LongFast" >> toggle Precise Location to on >> press Save >> press Send.

Looks to me like Metastatic lets you pick different levels of privacy for every mesh channel your apart of. After doing this you may need power cycle, reset the location one last time.

**Comment by kn5g** [comment]
Reading through these comments I assume this issue has not been resolved? I am having similar issues with HelTec, RakWireless, and now SenseCap nodes, all running 2.5 or later firmware. I have used the iOS client, the Android client, and the Web Interface client to try and get these nodes to report their precise locations over the Mesh, with no luck. I just got the Python client set up on my windows computer and I will see if I have any better luck using it to configure the nodes to report their precise locations. When I was using the Android client to program a RakWireless node I was able to enter and save the LAT & LON values manually. However after a minute or so the values were changed in the node and it was reporting the same, non-precise location as it had been before. This seems to point to the firmware recalculating the LAT & LON values, (with an error) and using those values as the node’s position report. This certainly appears to be a firmware bug that needs to be looked at.

**Comment by bbrodka2** [comment]
Same here, GPS position keeps changing from what I manually set it to. tried Bluetooth, remote admin latest beta, rak, seeed, heltec, ios, web, android.  No luck no matter what I try

**Reply by kn5g** [reply]
I’ve never found nor heard an “official” explanation but I think it has to do with the Apple or iOS application. If I configure and view my nodes and the Mesh Map through an Android device I can set the position precision to precise in the channel zero configuration page and it stays set. The nodes report their correct, precise positions on the mesh map. Setting precise in channel zero configuration isn’t available in the iOS application for whatever reason. Kind of makes the iOS application useless if you want to see the node’s precise location.

**Reply by Stephan-2025** [reply]
I‘ve the issue too on Heltec v3 and T114

**Comment by rjDipcord** [comment]
So I have done some testing and I believe I understand the core of the issue here... 

**The fixed coordinates are governed by the precision set on the _primary_ channel.** 

That is, if you have LongFast as a primary and set the precision to anything other than precise, the fixed position data will be adjusted to fit whatever precision is set there - no matter what precision is set on alternate channels. When you have multiple channels and some of which you wish to have precise location turned on for, the primary channel must **also** have precise location turned on.

I tested this tonight when It was brought to my attention by a local mesh user. It would seem that the correct way to achieve imprecise locations on LongFast, and precise ones on your private channels would be to simply use a private channel as your primary. Note: the LoRa frequency slot will need manually reset to 20 after doing so.

I found this to be true on a Heltec V3 and a T-Beam Supreme.
